rclcpp/src/rclcpp/executors/static_single_threaded_executor.cpp

```diff
@@ -50,6 +51,34 @@ StaticSingleThreadedExecutor::spin()
   }
 }
 
+void
+StaticSingleThreadedExecutor::spin_some(std::chrono::nanoseconds max_duration)
+{
+  // Make sure the entities collector has been initialized
+  if (!entities_collector_->is_init()) {
+    throw std::runtime_error("spin_some() called without initializing the entities collector");
+  }
+
+  if (spinning.exchange(true)) {
+    throw std::runtime_error("spin_some() called while already spinning");
+  }
+  RCLCPP_SCOPE_EXIT(this->spinning.store(false); );
+
+  if (rclcpp::ok(context_) && spinning.load()) {
+    // Refresh wait set, wait for work and execute ready executables
+    entities_collector_->refresh_wait_set(max_duration);
```

Comment:
if i am not mistaken, i was expecting `max_duration` is time to spend executing work, not to wait on wait_set. and test would be also ideal.

Reply:
True, I have updated the implementation and added the `StaticSingleThreadedExecutor` to the spin tests.